events-api/src/modules/events/events.repository.ts

```diff
@@ -16,6 +20,130 @@ export class EventsRepository {
             events
     `);
 
-    return result;
-  }
+        return result;
+    }
+
+    static createEvent(event: EventsCreateDto): Promise<EventModel[]> {
+        const { eventName, eventDescription, eventLocation, eventDate } =
+            event;
+        const eventId = uuid();
+
+        const result = db.execute<EventModel>(
+            `
+            INSERT INTO events
+            (event_id, event_name, event_description, event_location, event_date)
+            VALUES (?, ?, ?, ?, ?)
+            RETURNING *
+        `,
+            [
+                eventId,
+                eventName,
+                eventDescription,
+                eventLocation,
+                eventDate
+            ]
+        );
+
+        return result;
+    }
+
+    static async getEvent(id: string): Promise<EventModel[]> {
+        const result = db.execute<EventModel>(
+            `
+            SELECT ${this.aliases}
+            FROM events
+            WHERE event_id = ?
+            `,
+            [id]
+        );
+
+        return result;
+    }
+
+    static async updateEvent(
+        eventId: string,
+        event: EventsUpdateDto
+    ): Promise<EventModel[]> {
+        const result = await EventsRepository.getEvent(eventId);
+
+        if (!result.length) {
+            throw new Error("Event Not Found");
```

Comment:
Such checks should be done on the service level

Reply:
moved the check to service level